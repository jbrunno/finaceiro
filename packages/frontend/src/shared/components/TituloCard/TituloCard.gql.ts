import { gql } from '@apollo/client';

export const CLIENTE_TITULO_FRAGMENT = {
  tituloQuestionario: gql`
    fragment CLIENTE_TITULO_FRAGMENT on TituloQuestionario {
      contaAnteriorFalencia
      observacao
      siteRegistro
      tipoAtivoInvestido
      tipoConta
      valorDepositos
    }
  `,
};

export type TituloQuestionario = {
  siteRegistro?: string | null;
  tipoConta?: string | null;
  contaAnteriorFalencia: boolean;
  tipoAtivoInvestido?: string | null;
  valorDepositos?: string | null;
  observacao?: string | null;
};
