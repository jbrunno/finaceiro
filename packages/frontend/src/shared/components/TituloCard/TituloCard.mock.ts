import { faker } from '@faker-js/faker';

export const tituloId = faker.datatype.uuid();
export const assinatura = true;

export const tituloQuestionario = {
  siteRegistro: 'siteRegistro',
  tipoConta: 'tipoConta',
  contaAnteriorFalencia: true,
  tipoAtivoInvestido: 'tipoAtivoInvestido',
  valorDepositos: 'valorDepositos',
  observacao: 'observacao',
};
