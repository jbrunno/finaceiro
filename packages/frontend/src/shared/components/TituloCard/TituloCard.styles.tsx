import { styled, css } from '@mui/material/styles';
import { Card, CardActions, CardContent } from '@frontend/components';

export const CardActionsStyled = styled(CardActions)`
  justify-content: center;
  border-top: 1px solid ${({ theme }) => theme.palette.divider};
`;

export const CardStyled = styled(Card)`
  display: grid;
  grid-template-rows: auto 1fr auto;
`;

export const CardContentStyled = styled(CardContent)(
  ({ theme }) => css`
    border-top: 1px solid ${theme.palette.divider};
    display: grid;
    grid-gap: ${theme.fn.spacing(4)};
    padding-top: 0;
  `,
);
