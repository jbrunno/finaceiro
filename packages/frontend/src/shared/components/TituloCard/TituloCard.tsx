import {
  Button,
  CardHeader,
  CardSkeleton,
  CardText,
  Chip,
  Grid,
  IconButton,
  Menu,
  MenuItem,
  RenderComponent,
  RenderComponents,
  Typography,
} from '@frontend/components';
import { MoreVert as MoreVertIcon } from '@mui/icons-material';
import { TituloDialog } from '../TituloDialog/TituloDialog';
import { TituloQuestionarioDialog } from '../TituloQuestionarioDialog/TituloQuestionarioDialog';
import { CLIENTE_TITULO_FRAGMENT, TituloQuestionario } from './TituloCard.gql';
import {
  CardActionsStyled,
  CardContentStyled,
  CardStyled,
} from './TituloCard.styles';
import { AdicionarAssinaturaDialog } from './components/AdicionarAssinaturaDialog/AdicionarAssinaturaDialog';
import { RemoverAssinaturaDialog } from './components/RemoverAssinaturaDialog/RemoverAssinaturaDialog';
import { useTituloCard } from './hooks/useTituloCard';

type TituloData = {
  screen: string;
  titulo: {
    id?: string | null;
    tituloQuestionario?: TituloQuestionario;
  } | null;
  loading: boolean;
  assinatura?: boolean;
};

type TituloCardProps = Omit<TituloData, 'tituloId' | 'tituloQuestionario'> & {
  className?: string;
  loading: boolean;
  refetch: (target: 'titulos') => void;
};

export function TituloCard({
  screen,
  titulo,
  className,
  assinatura,
  refetch,
  loading,
}: TituloCardProps) {
  if (loading) return <CardSkeleton />;

  const { id: tituloId, tituloQuestionario } = titulo || {};

  const {
    anchorEl,
    setAnchorEl,
    addAssinaturaIsOpen,
    setAddAssinaturaAsOpen,
    setTituloModalAsOpen,
    tituloModalIsOpen,
    setQuestionarioAsOpen,
    questionarioIsOpen,
    removeAssinaturaIsOpen,
    setRemoveAssinaturaAsOpen,
  } = useTituloCard();

  return (
    <>
      <CardStyled className={className}>
        <CardHeader
          title={
            <Grid
              container
              alignItems="start"
              justifyContent="space-between"
              px={1}
            >
              <Typography variant="h5">Informações do título</Typography>
              {assinatura && <Chip label="Assinado" />}
            </Grid>
          }
          action={
            <>
              <RenderComponent screen={screen}>
                <IconButton
                  id="signature-button"
                  aria-controls={anchorEl ? 'signature-menu' : undefined}
                  aria-haspopup="true"
                  aria-expanded={anchorEl ? 'true' : undefined}
                  onClick={(event) => setAnchorEl(event.currentTarget)}
                  size="small"
                  w-perm="TITULO_ASSINATURA"
                  disabled={!titulo}
                >
                  <MoreVertIcon />
                </IconButton>
              </RenderComponent>
              <Menu
                id="signature-menu"
                anchorEl={anchorEl}
                open={!!anchorEl}
                onClose={() => setAnchorEl(null)}
                MenuListProps={{
                  'aria-labelledby': 'signature-button',
                }}
              >
                <MenuItem
                  onClick={() => {
                    setAnchorEl(null);
                    return !assinatura
                      ? setAddAssinaturaAsOpen(true)
                      : setRemoveAssinaturaAsOpen(true);
                  }}
                >
                  {!assinatura ? 'Adicionar assinatura' : 'Remover assinatura'}
                </MenuItem>
              </Menu>
            </>
          }
        />
        <CardContentStyled>
          <CardText
            label="Tipo de conta"
            text={tituloQuestionario?.tipoConta || '-'}
          />
          <CardText
            label="Tipo de ativos utilizados"
            text={tituloQuestionario?.tipoAtivoInvestido || '-'}
          />
          <CardText
            label="Valor aproximado dos depósitos"
            text={tituloQuestionario?.valorDepositos || '-'}
          />
        </CardContentStyled>
        <CardActionsStyled>
          <Button
            disabled={!tituloQuestionario}
            variant="outlined"
            color="primary"
            onClick={() => setQuestionarioAsOpen(true)}
          >
            Ver detalhes
          </Button>
          <Button
            variant="contained"
            color="primary"
            onClick={() => setTituloModalAsOpen(true)}
          >
            {tituloQuestionario ? 'Editar dados' : 'Cadastrar'}
          </Button>
        </CardActionsStyled>
      </CardStyled>
      {tituloQuestionario && (
        <TituloQuestionarioDialog
          open={questionarioIsOpen}
          onClose={() => setQuestionarioAsOpen(false)}
          tituloQuestionario={tituloQuestionario}
        />
      )}
      <TituloDialog
        open={tituloModalIsOpen}
        onSaved={() => setTituloModalAsOpen(false)}
        onClose={() => setTituloModalAsOpen(false)}
        tituloId={tituloId}
        tituloQuestionario={tituloQuestionario || null}
      />
      <RenderComponents screen={screen}>
        <AdicionarAssinaturaDialog
          open={addAssinaturaIsOpen}
          onClose={() => setAddAssinaturaAsOpen(false)}
          tituloId={tituloId}
          w-perm="TITULO_ADICIONAR_ASSINATURA"
        />
        <RemoverAssinaturaDialog
          tituloId={String(tituloId)}
          refetch={refetch}
          open={removeAssinaturaIsOpen}
          onClose={() => setRemoveAssinaturaAsOpen(false)}
          w-perm="TITULO_REMOVER_ASSINATURA"
        />
      </RenderComponents>
    </>
  );
}

TituloCard.fragments = CLIENTE_TITULO_FRAGMENT;
