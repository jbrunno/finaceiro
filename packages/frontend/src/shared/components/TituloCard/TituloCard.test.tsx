import { render } from '@test-utils/component-testing';
import { TituloCard } from './TituloCard';
import { tituloQuestionario, tituloId, assinatura } from './TituloCard.mock';

test('should render TituloCard', () => {
  const { container } = render(
    <TituloCard
      screen="test"
      loading={false}
      titulo={{
        tituloQuestionario,
        id: tituloId,
      }}
      assinatura={assinatura}
      refetch={jest.fn}
    />,
  );
  expect(container).toMatchSnapshot();
});
