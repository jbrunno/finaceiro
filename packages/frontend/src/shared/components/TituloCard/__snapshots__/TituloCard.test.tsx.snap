// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TituloCard 1`] = `
.emotion-2 {
  background-color: #fff;
  color: #3B3B41DE;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  overflow: visible;
  border-radius: 0.5rem;
  box-shadow: 0px 2px 4px -1px rgba(233, 233, 233, 0.8),0px 4px 5px rgba(233, 233, 233, 0.14),0px 1px 10px rgba(233, 233, 233, 0.12);
  background-color: #fff;
  display: grid;
  grid-template-rows: auto 1fr auto;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
}

.emotion-4 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-5 {
  margin: 0;
  font-weight: 700;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0;
  font-family: Lato,sans-serif;
  display: block;
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.emotion-7 {
  margin: 0;
  font-weight: 700;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0;
  font-family: Lato,sans-serif;
}

.emotion-8 {
  max-width: 100%;
  font-family: Lato,sans-serif;
  font-size: 0.8125rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 32px;
  color: #3B3B41DE;
  background-color: #00000014;
  border-radius: 16px;
  white-space: nowrap;
  -webkit-transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  cursor: default;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  padding: 0;
  vertical-align: middle;
  box-sizing: border-box;
}

.emotion-8.Mui-disabled {
  opacity: 0.38;
  pointer-events: none;
}

.emotion-8 .MuiChip-avatar {
  margin-left: 5px;
  margin-right: -6px;
  width: 24px;
  height: 24px;
  color: #616161;
  font-size: 0.75rem;
}

.emotion-8 .MuiChip-avatarColorPrimary {
  color: #fff;
  background-color: #2025EF;
}

.emotion-8 .MuiChip-avatarColorSecondary {
  color: #fff;
  background-color: #7B1FA2;
}

.emotion-8 .MuiChip-avatarSmall {
  margin-left: 4px;
  margin-right: -4px;
  width: 18px;
  height: 18px;
  font-size: 0.625rem;
}

.emotion-8 .MuiChip-icon {
  margin-left: 5px;
  margin-right: -6px;
  color: #616161;
}

.emotion-8 .MuiChip-deleteIcon {
  -webkit-tap-highlight-color: transparent;
  color: rgba(59, 59, 65, 0.26);
  font-size: 22px;
  cursor: pointer;
  margin: 0 5px 0 -6px;
}

.emotion-8 .MuiChip-deleteIcon:hover {
  color: rgba(59, 59, 65, 0.4);
}

.emotion-9 {
  overflow: hidden;
  text-overflow: ellipsis;
  padding-left: 12px;
  padding-right: 12px;
  white-space: nowrap;
}

.emotion-10 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  margin-top: -4px;
  margin-right: -8px;
  margin-bottom: -4px;
}

.emotion-12 {
  padding: 16px;
  border-top: 1px solid #0000001F;
  display: grid;
  grid-gap: 0.25rem;
  padding-top: 0;
}

.emotion-12:last-child {
  padding-bottom: 24px;
}

.emotion-15 {
  margin: 0;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.125rem;
  letter-spacing: 0.0106rem;
  font-family: Lato,sans-serif;
  color: #3B3B4199;
}

.emotion-16 {
  margin: 0;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
}

.emotion-26 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-top: 1px solid #0000001F;
}

.emotion-26>:not(:first-of-type) {
  margin-left: 8px;
}

.emotion-28 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 5px 15px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: 1px solid rgba(74, 49, 255, 0.5);
  color: #4A31FF;
  text-transform: none;
}

.emotion-28::-moz-focus-inner {
  border-style: none;
}

.emotion-28.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-28 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-28:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(74, 49, 255, 0.04);
  border: 1px solid #4A31FF;
}

@media (hover: none) {
  .emotion-28:hover {
    background-color: transparent;
  }
}

.emotion-28.Mui-disabled {
  color: #00000042;
  border: 1px solid #0000001F;
}

.emotion-29 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-31 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #4A31FF;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  text-transform: none;
}

.emotion-31::-moz-focus-inner {
  border-style: none;
}

.emotion-31.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-31 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-31:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #2025EF;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-31:hover {
    background-color: #4A31FF;
  }
}

.emotion-31:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-31.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-31.Mui-disabled {
  color: #00000042;
  box-shadow: none;
  background-color: #0000001F;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0 emotion-1 emotion-2"
  >
    <div
      class="MuiCardHeader-root emotion-3"
    >
      <div
        class="MuiCardHeader-content emotion-4"
      >
        <span
          class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title emotion-5"
        >
          <div
            class="MuiGrid-root MuiGrid-container emotion-6"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 emotion-7"
            >
              Informações do título
            </h5>
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault emotion-8"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium emotion-9"
              >
                Assinado
              </span>
            </div>
          </div>
        </span>
      </div>
      <div
        class="MuiCardHeader-action emotion-10"
      />
    </div>
    <div
      class="MuiCardContent-root emotion-11 emotion-12"
    >
      <div
        class="MuiBox-root emotion-13"
      >
        <span
          class="MuiTypography-root MuiTypography-caption emotion-14 emotion-15"
        >
          Tipo de conta
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-16"
        >
          tipoConta
        </p>
      </div>
      <div
        class="MuiBox-root emotion-13"
      >
        <span
          class="MuiTypography-root MuiTypography-caption emotion-14 emotion-15"
        >
          Tipo de ativos utilizados
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-16"
        >
          tipoAtivoInvestido
        </p>
      </div>
      <div
        class="MuiBox-root emotion-13"
      >
        <span
          class="MuiTypography-root MuiTypography-caption emotion-14 emotion-15"
        >
          Valor aproximado dos depósitos
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-16"
        >
          valorDepositos
        </p>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing emotion-25 emotion-26"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium emotion-27 emotion-28"
        tabindex="0"
        type="button"
      >
        Ver detalhes
        <span
          class="MuiTouchRipple-root emotion-29"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium emotion-27 emotion-31"
        tabindex="0"
        type="button"
      >
        Editar dados
        <span
          class="MuiTouchRipple-root emotion-29"
        />
      </button>
    </div>
  </div>
</div>
`;
